docs/sop/DEPLOYING_TO_PRODUCTION.md

```diff
@@ -0,0 +1,321 @@
+# Deploying to production for the Upstream projects
+
+This document is a SOP is for preparing the infrastructure used for the live production environments of GigaDB. 
+This document is for core team members who need to work on live production infrastructure.
+
+The base principles, tooling, configuration and provisioning approach are the same that used for developers environments on AWS.
+The implications are:
+* You must be familiar with the content of and have provisioned and deployed following the guidance of `docs/SETUP_CI_CD_PIPELINES.md` and `docs/SETUP_PROVISIONING.md`
+* This doc will focus only on the specificities of the Upstream projects, as you can refer to the above docs for everything else
+
+## Upstream projects
+
+There are two Upstream projects on Gitlab, `gigadb-website` and `alt-gigadb-website`.
+Together they represent an implementation of:
+* An high availability strategy called "Hot Stand-By"  [1]
+* A deployment strategy called "Blue/Green" [2]
+* A failover strategy, AWS calls "Portfolio failover" [3]
+
+The two infrastructures are meant to be identical and independent, and must run on two distinct regions. Each infrastructure has a role and they will swap roles upon blue/green deployment or incident failover.
+
+| Gitlab project     | AWS profile | Local checkout name | AWS region              | current role | current release | status |
+|--------------------| --- |---------------------|-------------------------| --- |-----------------|--------|
+| gigadb-website     | Upstream | gigadb-upstream     | ap-east-1 (HK)          | current production             | v4.3.4          | OK     |
+| alt-gigadb-website | UpstreamAlt | gigadb-alt-upstream | ap-southeast-2 (Sydney) | hot stand-by   | v0337-alt-upstream-beta          | OK     |
+
+>**Note**: this table needs updating after each deployment or incident failover
+
+
+
+[1] https://en.wikipedia.org/wiki/Hot_spare
+
+[2] https://en.wikipedia.org/wiki/Blue–green_deployment
+
+[3] https://aws.amazon.com/blogs/architecture/creating-an-organizational-multi-region-failover-strategy/
+
+## Configuring locally AWS for Upstream projects
+
+Normally, in `~/.aws/credentials`, you should already have a profile called `[GigaDB]` that you have used for creating developer environments on AWS.
+You now need two new profiles:
+* `[Upstream]`: for the `gigadb-website` project
+* `[UpstreamAlt]`: for the `alt-gigadb-website` project
+
+You populate these sections with AWS access keys you create in the IAM dashboard for the GigaDB user.
+
+```
+[Upstream]
+aws_access_key_id=<aws_access_key_id1 for Gigadb user>
+aws_secret_access_key=<aws_secret_access_key1 for Gigadb user>
+
+[UpstreamAlt]
+aws_access_key_id=<aws_access_key_id2 for Gigadb user>
+aws_secret_access_key=<aws_secret_access_key2 for Gigadb user>
```

Comment:
I don't have the access keys for `[UpstreamAlt]`. Put these in `cnhk-infra`? Also, update the name of the `[Upstream]` access keys already in `cnhk-infra` from `Gigadb_aws_access_key_id` to `Gigadb_upstream_aws_access_key_id`? Also `Gigadb_aws_secret_access_key` to `Gigadb_upstream_aws_secret_access_key` to distinguish them from the UpstreamAlt variables.